lib/rubygems/config_file.rb

```diff
@@ -210,22 +210,32 @@ def initialize(args)
       @hash = @hash.merge environment_config
     end
 
+    @hash.transform_keys! do |k|
+      if %w[backtrace bulk_threshold verbose update_sources cert_expiration_length_days
+            install_extension_in_lib ipv4_fallback_enabled].include?(k)
+        k.to_sym
+      else
+        k
+      end
+    end
+
     # HACK: these override command-line args, which is bad
     @backtrace                   = @hash[:backtrace]                   if @hash.key? :backtrace
     @bulk_threshold              = @hash[:bulk_threshold]              if @hash.key? :bulk_threshold
-    @home                        = @hash[:gemhome]                     if @hash.key? :gemhome
-    @path                        = @hash[:gempath]                     if @hash.key? :gempath
-    @update_sources              = @hash[:update_sources]              if @hash.key? :update_sources
     @verbose                     = @hash[:verbose]                     if @hash.key? :verbose
-    @disable_default_gem_server  = @hash[:disable_default_gem_server]  if @hash.key? :disable_default_gem_server
-    @sources                     = @hash[:sources]                     if @hash.key? :sources
+    @update_sources              = @hash[:update_sources]              if @hash.key? :update_sources
+    # concurrent_downloads
```

Comment:
Is the above comment intentional?

Reply:
Yes. I'm not sure why `concurrent_download` is not handled this place. We should fix that for `Gem.configuration.concurrent_downloads`.

But It's not this PR scope.

Reply:
So basically this comment is acting as a TODO note? If that's the case, I'd make that a bit more clear like `TODO: Review why concurrent_downloads is not handled here`.

Reply:
I see. I did update that.